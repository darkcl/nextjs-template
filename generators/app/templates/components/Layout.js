import React from 'react';
import Router from 'next/router';
import NProgress from 'nprogress';

import Header from './Header';
import Footer from './Footer';

import config from '../configs/config';

// progress bar
Router.onRouteChangeStart = () => {
  NProgress.start();
};
Router.onRouteChangeComplete = () => NProgress.done();
Router.onRouteChangeError = () => NProgress.done();

const Layout = props => [
  <Header key="0" imagePath={config.paths.images}/>,
  <div key="1" className="container">
    {props.children}
  </div>,
  <Footer key="2" />,
];

export default Layout;
