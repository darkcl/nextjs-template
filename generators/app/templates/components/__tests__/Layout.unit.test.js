import React from 'react';
import toJson from 'enzyme-to-json';
import { render } from 'enzyme';

import Layout from '../Layout';

describe('<Layout />', () => {
  it('should render correctly with no children', () => {
    const layout = render(<Layout/>);
    expect(toJson(layout)).toMatchSnapshot();
  });

  it('should render correctly with some children', () => {
    const layout = render(<Layout>
      <div>Hello World 1</div>
      <p>Hello World 2</p>
      <span>Hello World 3</span>
    </Layout>);
    expect(toJson(layout)).toMatchSnapshot();
  });
});
