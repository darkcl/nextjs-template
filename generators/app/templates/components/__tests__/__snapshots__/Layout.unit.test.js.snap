// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render correctly with no children 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          src="undefined/static/47cd76e43f74bbc2e1baaf194d07e1fa/images/favicon.png"
        />
      </a>
      <span>
        header.title
      </span>
    </div>
  </div>,
  <div
    class="container"
  />,
  <div
    class="footer"
  >
    <div
      class="container"
    >
      footer.text
    </div>
  </div>,
]
`;

exports[`<Layout /> should render correctly with some children 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          src="undefined/static/47cd76e43f74bbc2e1baaf194d07e1fa/images/favicon.png"
        />
      </a>
      <span>
        header.title
      </span>
    </div>
  </div>,
  <div
    class="container"
  >
    <div>
      Hello World 1
    </div>
    <p>
      Hello World 2
    </p>
    <span>
      Hello World 3
    </span>
  </div>,
  <div
    class="footer"
  >
    <div
      class="container"
    >
      footer.text
    </div>
  </div>,
]
`;
